client/context/broadcast.go

```diff
@@ -29,6 +31,27 @@ func (ctx CLIContext) BroadcastTx(txBytes []byte) (res sdk.TxResponse, err error
 	return res, err
 }
 
+func CheckTendermintError(err error) *sdk.TxResponse {
```

Comment:
Let's add a godoc here 👍 

Also, it'll be helpful for clients to have the tx hash also be included as part of the response. You can do this by simply doing:

```
import 	"github.com/tendermint/tendermint/crypto/tmhash"

tmhash.Sum(txBytes)
```